pkgs/by-name/re/realcugan-ncnn-vulkan/package.nix

```
@@ -0,0 +1,70 @@
+{
+  lib,
+  stdenv,
+  fetchzip,
+  fetchFromGitHub,
+  cmake,
+  vulkan-headers,
+  vulkan-loader,
+  glslang,
+  libwebp,
+  ncnn,
+}:
+stdenv.mkDerivation rec {
+  pname = "realcugan-ncnn-vulkan";
+  version = "20220728";
+
+  src = fetchFromGitHub {
+    owner = "nihui";
+    repo = "realcugan-ncnn-vulkan";
+    rev = version;
+    hash = "sha256-P3Y1B8m1+mpFinacwnvBE2vU150jj6Q12IS6QYNRZ6A=";
+  };
+  sourceRoot = "${src.name}/src";
+
+  models = fetchzip {
+    url = "https://github.com/nihui/realcugan-ncnn-vulkan/releases/download/20220728/realcugan-ncnn-vulkan-20220728-ubuntu.zip";
+    sha256 = "sha256-71C6taL2Zr1exG5HEXOLy1j9ZMKgkMJjTgNi2hiA7xk=";
+  };
+
+  patches = [
+    ./cmakelists.patch
+    ./models_path.patch
+  ];
+
+  cmakeFlags = [
+    "-DUSE_SYSTEM_NCNN=1"
```

```suggestion
    (lib.cmakeBool "USE_SYSTEM_NCNN" true)
```
Similar for others, the last can use lib.cmakeFeature

---

Thanks, made the changes